daemon/emer-machine-id-provider.c

```diff
@@ -291,24 +378,64 @@ emer_machine_id_provider_get_id (EmerMachineIdProvider *self,
   EmerMachineIdProviderPrivate *priv =
     emer_machine_id_provider_get_instance_private (self);
 
-  static gboolean id_is_valid = FALSE;
-  G_LOCK_DEFINE_STATIC (id_is_valid);
-
-  G_LOCK (id_is_valid);
-  if (!id_is_valid)
+  if (!priv->id_is_valid)
     {
       if (read_machine_id (self))
         {
-          id_is_valid = TRUE;
+          priv->id_is_valid = TRUE;
         }
       else
         {
-          G_UNLOCK (id_is_valid);
           return FALSE;
         }
     }
-  G_UNLOCK (id_is_valid);
 
   uuid_copy (machine_id, priv->id);
   return TRUE;
 }
+
+static gboolean
+write_tracking_id_file (const gchar  *path,
+                        GError      **error)
+{
+  uuid_t override_machine_id;
+  gchar serialized_override_machine_id[UUID_SERIALIZED_LEN];
+  g_autoptr(GFile) file = g_file_new_for_path (path);
+  g_autoptr(GFile) directory = g_file_get_parent (file);
+  g_autoptr(GError) local_error = NULL;
+
+  uuid_clear (override_machine_id);
+  uuid_generate (override_machine_id);
+  uuid_unparse (override_machine_id, serialized_override_machine_id);
+
+  if (!g_file_make_directory_with_parents (directory, NULL, &local_error))
+    {
+      if (!g_error_matches (local_error, G_IO_ERROR, G_IO_ERROR_EXISTS))
+        {
+          g_propagate_error (error, g_steal_pointer (&local_error));
+          return FALSE;
+        }
+
+      g_clear_error (&local_error);
+    }
+
+  if (!g_file_set_contents (path, serialized_override_machine_id, -1, error))
+    return FALSE;
+
+  return TRUE;
+}
+
+gboolean
+emer_machine_id_provider_reset_tracking_id (EmerMachineIdProvider  *self,
+                                            GError                **error)
+{
+  EmerMachineIdProviderPrivate *priv =
+    emer_machine_id_provider_get_instance_private (self);
+
+  if (!write_tracking_id_file (priv->override_path, error))
+    return FALSE;
+
+  g_message ("EmerMachineIdProvider: Will reload from:\n%s", priv->override_path);
+
+  priv->id_is_valid = FALSE;
```

Comment:
This function needs to `return TRUE` on the happy path. I'd hope the compiler would issue a warning.

Reply:
It does not, unfortunately, I have noticed :(